crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -361,4 +403,23 @@ module sui::staking_pool {
 
         staked_amount + reward_withdraw_amount
     }
+
+    #[test_only]
```

Comment:
Unrelated side note, but that might make sense if it makes testing factor:I think we should provide a test-only polymorphic `destroy` in the `test_utils` module. Handwriting these test destructors gets really painful, and if one of your deps forgets to provide one, it can be impossible to test your code.

Reply:
I like it! I think for this PR I'm going to go back to a `Table` and add the `test_only` function to destroy a table, but will do a quick followup afterwards adding this `destroy` function + removal of the different test only destroy functions that we have laying around :)

Reply:
Wound up implementing this `destroy` function and this PR is now rebased on it. It makes it much nicer!